lightkurve/lightcurve.py

```diff
@@ -1085,3 +1086,153 @@ def opt_depth(to, width, data):
+            except:
+                raise TypeError("Object is not a LightCurve instance")
+
+    def __len__(self):
+        return len(self.data)
+
+    def _ids(self):
+        """
+        Returns the kepler_ids of all the lightcurves as a dict_keys obj.
+        """
+        return self.data.keys()
+
+    def __getitem__(self, kep_id):
+        """
+        Returns the lightcurve associated with the kepler_id. 
+        """
+        try: 
+            return self.data[kep_id]
+        except:
+            raise ValueError('No LightCurve for ' + kep_id)
+
+    def append(self, lc):
+        try:
+            self.data[lc.keplerid] = lc
+        except:
+            raise TypeError("Input is not a lightcurve")
+
+    def __repr__(self):
+        result = ""
+        for lightcurve in self.data.values():
+            result += lightcurve.__repr__() + "\n"
+        return result
+
+    def plot(self):
+        """
+        Plot a collection of LightCurves. Random colors are assigned to each plot.
+        """
+        for i, k_id in enumerate(self.data):
+            rand_color = np.random.rand(3)
+            if i == 0:
+                #TODO: what if longest axis is not the first item?
+                axis = self.data[k_id].plot(color=rand_color, linestyle='-',label=k_id)
+            else:
+                self.data[k_id].plot(ax=axis, color=rand_color, linestyle='-', label=k_id)
+
+
+    def pca(self):
+        '''Creates the Principle Components of a collection of LightCurves
+        '''
+        raise NotImplementedError('Should be able to run a PCA on a collection.')
+
+class TargetPixelFileCollection(object):
+    """
+    Collects multiple TPF objects together with helpful functions.
+    """
+    def __init__(self, targetPixelFiles):
+        try:
+            tpfs = np.asarray(targetPixelFiles)
+        except:
+            raise TypeError("Unable to parse input")
+        self.data = {}
+        for TPF in tpfs:
+            try:
+                if TPF.keplerid:
+                    self.data[TPF.keplerid] = TPF
+            except:
+                raise TypeError("Object is not a TargetPixelFile instance")
+
+    def __len__(self):
+        return len(self.data)
+
+    def _ids(self):
+        """
+        Returns the kepler_ids of all the lightcurves as a dict_keys obj.
+        """
+        return self.data.keys()
+
+    def __getitem__(self, kep_id):
+        """
+        Returns the lightcurve associated with the kepler_id. 
+        """
+        try: 
+            return self.data[kep_id]
+        except:
+            raise ValueError('No TPF for ' + kep_id)
+
+    def append(self, tpf):
+        try:
+            self.data[tpf.keplerid] = tpf
+        except:
+            log.warning("Input is not a TPF")
+
+    def __repr__(self):
+        result = ""
+        for TPF in self.data.values():
+            result += TPF.__repr__() + "\n"
+        return result
+
+    def plot(self):
```

Comment:
Same as above, let's have `ax` and `**kwargs` to pass to matplotlib so that the function behaves just like `tpf.plot()`. (You'll have to be careful with vmin and vmax if you set them in this function and also pass them as kwargs.)